scripts/ice-road-pipeline.py

```diff
@@ -124,15 +164,170 @@
     snowon = rio.open_rasterio(snow_tif, masked=True) 
     snowon_matched = snowon.rio.reproject_match(snowoff)
     snowdepth = snowon_matched - snowoff
+
+    # Write reproject-match SnowDepth and SnowOn to disk
     snowdepth.rio.to_raster(snow_depth_path)
+    snowon_matched.rio.to_raster(snow_tif)
 
     # difference two rasters to find canopy height
     ref_dem_path = join(results_dir, 'dem.tif')
     canopy_fp = join(ice_dir, f'{basename(in_dir)}-canopyheight.tif')
     canopy = rio.open_rasterio(canopy_tif, masked=True) 
     matched = canopy.rio.reproject_match(snowoff)
     canopyheight = matched - snowoff
-    # mask snow depth from vegetation
-    canopyheight = canopyheight.where((canopyheight > snowon + 0.1) | (snowon.isnull()))
+
+    # mask snow depth from vegetation (null=0)
+    canopyheight = canopyheight.where((canopyheight > snowdepth + 0.1) | (snowdepth.isnull()), other=0)
+    canopyheight.rio.to_raster(canopy_fp)
+
+    # Command for testing...
+    # python ice-road-pipeline.py /Users/brent/Code/ice-road-copters/data/feb9/mcs/ -e /Users/brent/Code/LIDAR/data/QSI_snowfree.tif -a /Users/brent/Code/ice-road-copters/ASP/ -s /Users/brent/Code/ice-road-copters/transform_area/hwy_21/hwy21_utm_edit_v3.shp -r /Users/brent/Code/ice-road-copters/transform_area/Eagle/hill_rd_55_intersection.shp -i /Users/brent/Documents/MCS/LIDAR/IMU/20230209_MCS_precise_scanner_bw.txt -d True -c /Users/brent/Code/LIDAR/data/fl_230210_002840/20230209_extraBytes-230210_002840_Scanner_1.las -k 0.1956
+    if shp_fp_rfl:
+
+        # Make sure R is installed
+        proc = Popen(["which", "R"],stdout=PIPE,stderr=PIPE)
+        exit_code = proc.wait()
+        if exit_code != 0:
+            raise Exception("Please install R on this system and packages: raster, lidR, rlas, dplyr, readr, terra,sf, data.table.")
```

Comment:
Also this could be an ImportError not that it matters too much.

Reply:
With this being used less frequently in our group I agree with your comment it would be best to keep it R installation separate, and rely on user installing R packages independently. Also, from my limited experience with R it seems there are more package conflicts that arise as compared to python, and so that could be another reason to avoid automating a solution that may be difficult to manage. I'll consider the ImportError!